src/smoosh/src/smoosh_channel.erl

```diff
@@ -65,60 +80,84 @@ close(ServerRef) ->
 flush(ServerRef) ->
     gen_server:call(ServerRef, flush).
 
+is_key(ServerRef, Key) ->
+    gen_server:call(ServerRef, {is_key, Key}).
+
+is_activated(ServerRef) ->
+    gen_server:call(ServerRef, is_activated).
+
+persist(ServerRef) ->
+    gen_server:call(ServerRef, persist).
+
 % gen_server functions.
 
 init(Name) ->
-    schedule_unpause(),
     erlang:send_after(60 * 1000, self(), check_window),
-    {ok, #state{name = Name}}.
+    process_flag(trap_exit, true),
+    Waiting = smoosh_priority_queue:new(Name),
+    State = #state{name = Name, waiting = Waiting, paused = true, activated = false},
+    ok = gen_server:cast(self(), init),
```

Comment:
Now the `handle_cast(init, ...` function don't do much, so we can remove it and schedule `start_recovery` event here. 

```
erlang:send_after(?START_DELAY_IN_MSEC, self(), start_recovery),
```

Reply:
Removed `init`: https://github.com/apache/couchdb/pull/3766/commits/5c737f17e96d9a331fef2d6cfa8964b25ee8c53a#diff-113cd247cfe56b1aa882a37daa2856fd33bab2e2a6183034eb3a30088960dc13R107